internal/security/secretstore/vmkencryption.go

```diff
@@ -0,0 +1,291 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// ECCN: 5D002TSU
+//
+
+package secretstore
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/kdf"
+	"github.com/edgexfoundry/edgex-go/internal/security/pipedhexreader"
+	"github.com/edgexfoundry/edgex-go/internal/security/secretstoreclient"
+	"github.com/edgexfoundry/go-mod-secrets/pkg/token/fileioperformer"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+/*
+
+THE FOLLOWING DISCLAIMER IS REQUIRED TO BE CARRIED WITH THE BELOW CODE
+DO NOT REMOVE EXCEPT BY PERMISSION OF THE AUTHOR
+
+Vault Master Key Encryption Feature
+
+The purpose of this feature is to provide a secure way to unlock Vault
+without requiring human intervention to supply the Vault Master Key.
+
+This feature requires a sufficiently secure source of randomness in order
+to unlock the Vault. This randomness should have at least 256 bits of entropy
+and be backed by hardware secure storage such as a TPM or TEE or a hardware
+security token.  If using a TPM, the secret should be bound to platform
+configuration registers to attest the system state.
```

Comment:
> So what you're saying is that when using a TPM, a secret of at least 256 bits of entropy should be sealed to the TPM using a PCR-based policy? I suppose the actual policy (i.e. which PCRs) is left up to the implementer of the IKM?

Yes.

> How is the secret protected from other processes on the system (i.e. if the secret is bound using a PCR-based policy, as long as the PCR values on boot match the PCR values when the secret was sealed, the secret can be read by any application on the system)? Sometimes this is handled by extending one or more secrets after something has been unsealed, but this approach would cause restarts of EdgeX to fail (i.e. the IKM application must always return the same secret given an identical set of command-line options/inputs).

Arguably if one cannot prevent he execution of malicious code on the system this mechanism will do little.  Poisoning the PCR to prevent re-reads would have the above stated effect (only have one chance per boot to unseal Vault).  An implementation might choose to do something in between, such as reading the value out of a TPM, storing it in a kernel keyring, and then poisoning the PCR, and rely on user/group permissions to prevent access from then on.

The scope of the implementation is providing a mechanism to encrypt `resp-ini.json` using a key that is ultimately not stored in plaintext on media.